### PR TITLE
MM-12495 Removing some flux store usage.

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

**Comment by crspeller** [comment]
@lindy65 Can you do a quick smoke test of team and channel sidebars and the RHS sidebar on webapp mobile view?

File: components/rhs_header_post/rhs_header_post.jsx [hunk 1]
```diff
@@ -97,16 +91,6 @@ export default class RhsHeaderPost extends React.Component {
             );
             break;
         default:
```
Comment (lieut-data) [inline components/rhs_header_post/rhs_header_post.jsx]: remove `default` block altogether?

File: components/sidebar/header/sidebar_header.jsx [hunk 2]
```diff
@@ -5,28 +5,43 @@ import PropTypes from 'prop-types';
...
+        teamDisplayName: PropTypes.string.isRequired,
+        teamDescription: PropTypes.string.isRequired,
+        teamName: PropTypes.string.isRequired,
+        teamType: PropTypes.string.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        showTutorialTip: PropTypes.bool.isRequired,
+    };
+
+    static defaultProps = {
+        teamDisplayName: '',
```
Comment (lieut-data) [inline components/sidebar/header/sidebar_header.jsx]: I know it's just being relocated, but should we drop the `isRequired` for these props?

File: components/team_sidebar/team_sidebar_controller.jsx [hunk 3]
```diff
@@ -29,130 +30,37 @@ export default class TeamSidebar extends React.Component {
...
-        TeamStore.removeChangeListener(this.onChange);
-        TeamStore.removeUnreadChangeListener(this.onChange);
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (!this.state.isMobile) {
-            $('.team-wrapper').perfectScrollbar();
-        }
-
-        // reset the scrollbar upon switching teams
```
Comment (lieut-data) [inline components/team_sidebar/team_sidebar_controller.jsx]: Is this behaviour preserved somehow?

File: components/sidebar/header/sidebar_header.jsx (hunk 2 above)
Comment (crspeller) [reply components/sidebar/header/sidebar_header.jsx]: Just removed the defaults. That's probably better as they really are required.

File: components/team_sidebar/team_sidebar_controller.jsx (hunk 3 above)
Comment (crspeller) [reply components/team_sidebar/team_sidebar_controller.jsx]: No, it didn't work before anyway. 
Also if you have to scroll the team sidebar our UI is broken anyway.

File: components/sidebar/header/dropdown/sidebar_header_dropdown.jsx [hunk 4]
```diff
@@ -1,19 +1,15 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import $ from 'jquery';
```
Comment (saturninoabril) [inline components/sidebar/header/dropdown/sidebar_header_dropdown.jsx]: 🎉

**Comment by crspeller** [comment]
@saturninoabril Yup we are removing it.

**Pull request merged**